util/signing/signer/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+pub mod cryptography;
+
+/// A collection of bytes.
+#[derive(Debug, Clone)]
+pub struct Bytes(pub Vec<u8>);
+
+/// A signature.
+#[derive(Debug, Clone)]
+pub struct Signature(pub Bytes);
+
+/// A public key.
+#[derive(Debug, Clone)]
+pub struct PublicKey(pub Bytes);
+
+/// Version of a key.
+/// Default mean the current key.
+#[derive(Debug, Clone, Default)]
+pub struct KeyVersion(pub String);
+
+/// Id that identify a Key.
+#[derive(Debug, Clone)]
+pub struct KeyId(pub String);
+
+/// Errors thrown by Signer
+#[derive(Debug, thiserror::Error)]
+pub enum SignerError {
+	#[error("Error during signing : {0}")]
```

Comment:
@musitdev Notes on error message styling: the message does not need to start with "Error" (we already know it's an error), be capitalized (as the message is often formatted into a larger message), or end with a dot. The colon should not be preceded with a space.